include/boost/math/special_functions/fourier_transform_daubechies_scaling.hpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Nick Thompson, Matt Borland, 2023
+ * Use, modification and distribution are subject to the
+ * Boost Software License, Version 1.0. (See accompanying file
+ * LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_MATH_SPECIAL_FOURIER_TRANSFORM_DAUBECHIES_SCALING_HPP
+#define BOOST_MATH_SPECIAL_FOURIER_TRANSFORM_DAUBECHIES_SCALING_HPP
+#include <cmath>
+#include <complex>
+#include <iostream>
+#include <array>
+#include <limits>
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/tools/big_constant.hpp>
+
+namespace boost::math {
+
+namespace detail {
+
+// See the Table 6.2 of Daubechies, Ten Lectures on Wavelets.
+// These constants are precisely those divided by 1/sqrt(2), because otherwise we'd immediately just have to divide through by 1/sqrt(2):
+template<typename Real, size_t N>
+constexpr const std::array<Real, N> ft_daubechies_scaling_polynomial_coefficients() {
+    if constexpr (N==3) {
+	return std::array<Real, 3>{BOOST_MATH_BIG_CONSTANT(Real, std::numeric_limits<Real>::digits, 1.88186883113665472301331643028468183320710177910151845853383427363197699204347143889269703), BOOST_MATH_BIG_CONSTANT(Real, std::numeric_limits<Real>::digits, -1.08113883008418966599944677221635926685977756966260841342875242639629721931484516409937898), 0.199269998947534942986130341931677433652675790561089954894918152764320227250084833874126086};
+    }
+    else {
+       throw std::domain_error("Not yet implemented.");
```

Comment:
You can't throw in a constexpr function. If you replace this with an assertion it'll be fine.

Reply:
Simplest thing might be to remove the `size_t N` template parameter until it is actually ready for other values.

Anyway, I'll go back to my own thing now.  :)

Reply:
@jeremy-murphy : lol; all I had to do to get the correct values was just pump it through your polynomial division code.

@mborland : Thanks! Fixed.

Reply:
Actually you can throw in a constexpr function: but it will result in a compiler error if that branch is evaluated (which is what you want anyway).